src/include/duckdb/storage/buffer/buffer_pool.hpp

```diff
@@ -86,12 +92,32 @@ class BufferPool {
 	atomic<idx_t> maximum_memory;
 	//! Eviction queue
 	unique_ptr<EvictionQueue> queue;
-	//! Total number of insertions into the eviction queue. This guides the schedule for calling PurgeQueue.
-	atomic<uint32_t> queue_insertions;
 	//! Memory manager for concurrently used temporary memory, e.g., for physical operators
 	unique_ptr<TemporaryMemoryManager> temporary_memory_manager;
 	//! Memory usage per tag
 	atomic<idx_t> memory_usage_per_tag[MEMORY_TAG_COUNT];
+
+	//! We trigger a purge of the eviction queue every INSERT_INTERVAL insertions
+	constexpr static idx_t INSERT_INTERVAL = 4096;
+	//! We multiply the base purge size by this value.
+	constexpr static idx_t PURGE_SIZE_MULTIPLIER = 2;
+	//! We multiply the purge size by this value to determine early-outs. This is the minimum queue size.
+	//! We never purge below this point.
+	constexpr static idx_t EARLY_OUT_MULTIPLIER = 4;
+	//! We multiply the approximate alive nodes by this value to test whether our total dead nodes
+	//! exceed their allowed ratio. Must be greater than 1.
+	constexpr static idx_t ALIVE_NODE_MULTIPLIER = 4;
+
+	//! Total number of insertions into the eviction queue. This guides the schedule for calling PurgeQueue.
+	atomic<idx_t> evict_queue_insertions;
+	//! Total dead nodes in the eviction queue. There are two scenarios in which a node dies: (1) we destroy its block
+	//! handle, or (2) we insert a newer version into the eviction queue.
+	atomic<idx_t> total_dead_nodes;
+	//! Whether a queue purge is currently active. Only lets a single thread enter the purge phase.
+	atomic<bool> purge_active;
```

Comment:
Could we replace this with a regular mutex, using `try_lock`


https://en.cppreference.com/w/cpp/thread/mutex/try_lock#:~:text=Tries%20to%20lock%20the%20mutex,locked%20by%20any%20other%20thread.